index.js

```diff
@@ -32,6 +37,59 @@ module.exports = {
         ],
       },
     },
+    delete: {
+      validation: {
+        function: 'checkboxValidation',
+        on: 'change',
+      },
+      script: {
+        function: 'fileScript',
+        on: 'change',
+      },
+      type: 'checkbox',
+      label: 'Delete file',
+      options: [
+        { label: 'Delete file',
+          value: 'true',
+        },
+      ],
+      settings: {
+        classes: {
+          hide: 'file--uploaded',
+        },
+      },
+    },
   },
-  html: '<label for="{{file.id}}">{{file.label}}</label><input type="{{file.type}}" id="{{file.id}}" name="{{file.name}}" value="{{file.value}}" {% if settings.types %}{%set comma = joiner() %}accept="{% for type in settings.type %}{{comma()}}{{type}}{% endfor %}{% endif %}" />',
+  html: `<label for="{{file.id}}">{{file.label}}</label>
+    {% set classes = '' %}
+    {% set filepath = '/files' %}
+
+    {% if storage.public %}
+      {% set filepath = storage.public %}
+    {% endif %}
+
+    {% if file.value %}
+      {% set classes = delete.settings.classes.hide %}
+
+      {# regex: if file type is image #}
+      {% set imgregex = r/image.*/g %}
+
+      <a href="{{filepath}}{{file.value.relative}}">
+        {% if imgregex.test(file.value.type) %}
+          <img src="{{filepath}}{{file.value.relative}}">
+        {% else %}
+          {{file.value.relative.substring(file.value.relative.lastIndexOf('/') + 1)}}
+        {% endif %}
+      </a>
+      <div class="file--delete">
+        {% for option in delete.options %}
+          <input type="{{delete.type}}" name="{{delete.name}}" id="{{delete.id}}--{{loop.index}}" value="{{option.value}}" {% if delete.value %}{% if option.value in delete.value %}checked{% endif %}{% endif %}>
+          <label for="{{delete.id}}--{{loop.index}}">{{option.label}}</label>
+        {% endfor %}
+      </div>
+    {% endif %}
+    <div class="{{classes}}">
```

Comment:
Don't fully understand why this gets generated instead of it being hard coded like the rest of the classes